src/storage/documentService/documentService.js

```diff
@@ -7,21 +7,24 @@ const { put, get, remove } = require("../dynamoDb");
 const DEFAULT_TTL = 60 * 60; // 1 Hour
 const MAX_TTL = 60 * 60 * 24 * 30; // 30 Days
 
-const putDocument = async (document, ttl = DEFAULT_TTL) => {
+const putDocument = async (document, id, ttl = DEFAULT_TTL) => {
   // TTL is handled by dynamoDb natively, this timestamp has to be UTC unixtime in seconds
   const created = Math.floor(Date.now() / 1000);
   // https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/time-to-live-ttl-how-to.html
   const params = {
     TableName: config.dynamodb.storageTableName,
     Item: {
-      id: uuid(),
+      id,
       document,
       created,
       ttl: created + ttl
     }
   };
-
-  return put(params).then(() => params.Item);
+  return put(params)
+    .then(() => params.Item)
+    .catch(e => {
```

Comment:
I was trying to hit the api without ttl value. Couldn`t identify the error. so handled it. Easy to track the error if passing some wrong value.

Reply:
.catch(e => { throw new Error(e) }) doesn't do anything though?
.catch(e) only executes when an error is throw in the promise, and if you simply re-throw the error then it's the same as not having the .catch because the error will surface regardless?

Reply:
ahh..okay ..my mistake.